src/transports/http.ts

```diff
@@ -192,21 +192,25 @@ export class HttpTransport<T extends HttpTransportGenerics> extends Subscription
 
   private async makeRequest(
     requestConfig: ProviderRequestConfig<T>,
-    adapterConfig: AdapterConfig<T['CustomSettings']>,
+    context: EndpointContext<T>,
   ): Promise<TimestampedProviderResult<T>[]> {
     try {
       // The key generated here that we pass to the requester is potentially very long, but we're not considering it an issue given that:
       //   - the requester will store values in memory, so we're not sending the string anywhere
       //   - there's no problems using very large strings as object keys
       //   - there should be a limit on the amount of subscriptions in the set
+      // Append endpoint name to key to avoid coalescing requests across different endpoints
+      const requestKey = `${context.endpointName}-${requestConfig.params
+        .map((p) => JSON.stringify(p))
+        .join('|')}`
```

Comment:
If we're going to add the endpoint, would it make sense to just use the cache key?

Reply:
Yup that's a much better idea